plugins/hls-eval-plugin/src/Ide/Plugin/Eval/CodeLens.hs

```diff
@@ -215,12 +215,12 @@ runEvalCmd plId st EvalParams{..} =
             mdlText <- moduleText _uri
 
             -- enable codegen for the module which we need to evaluate.
-            liftIO $ queueForEvaluation st nfp
-            liftIO $ setSomethingModified VFSUnmodified st [toKey NeedsCompilation nfp] "Eval"
-            -- Setup a session with linkables for all dependencies and GHCi specific options
-            final_hscEnv <- liftIO $ initialiseSessionForEval
-                                      (needsQuickCheck tests)
-                                      st nfp
+            final_hscEnv <- liftIO $ bracket_
+              (do queueForEvaluation st nfp
+                  setSomethingModified VFSUnmodified st [toKey IsEvaluating nfp] "Eval")
+              (do unqueueForEvaluation st nfp
+                  setSomethingModified VFSUnmodified st [toKey IsEvaluating nfp] "Eval")
```

Comment:
@pepeiborra do you think we need this `setSomethingModified` call after calling `unqueueForEvaluation`?